apps/keyboard/js/views/latin_candidate_panel_view.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+/* global CandidatePanelView */
+
+(function(exports) {
+/**
+ * CandidatePanelView handles the rendering of candidate panel, used for showing
+ * word suggestions.
+ */
+
+function LatinCandidatePanelView() {
+  CandidatePanelView.apply(this, arguments);
+}
+
+LatinCandidatePanelView.prototype = Object.create(CandidatePanelView.prototype);
+
+LatinCandidatePanelView.prototype.render = function render() {
+  var candidatePanel = document.createElement('div');
+  candidatePanel.setAttribute('role', 'group');
+  candidatePanel.dataset.l10nId = 'wordSuggestions2';
+
+  candidatePanel.classList.add('keyboard-candidate-panel');
+  if (this.className) {
+    candidatePanel.classList.add(this.className);
+  }
+
+  var dismissButton = document.createElement('div');
+  dismissButton.classList.add('dismiss-suggestions-button');
+  dismissButton.classList.add('hide');
+  dismissButton.setAttribute('role', 'button');
+  dismissButton.dataset.l10nId = 'dismiss2';
+  candidatePanel.appendChild(dismissButton);
+
+  this.dismissButton = dismissButton;
+
+  var dismissButtonView = {
+    element: dismissButton
+  };
+  this.viewManager.registerView({ isDismissSuggestionsButton: true },
+                                dismissButtonView);
+
+  var suggestionsContainer = document.createElement('div');
+  suggestionsContainer.classList.add('suggestions-container');
+  suggestionsContainer.setAttribute('role', 'listbox');
+  candidatePanel.appendChild(suggestionsContainer);
+
+  this.suggestionsContainer = suggestionsContainer;
+
+  this.element = candidatePanel;
+};
+
+LatinCandidatePanelView.prototype.showCandidates = function (candidates) {
+  if (!this.element) {
+    console.error('CandidatePanel is not ready yet!');
+    return;
+  }
+
+  this.dismissButton.classList.toggle('hide', (candidates.length <= 0));
+  this.suggestionsContainer.innerHTML = '';
+
+  candidates.forEach(function buildCandidateEntry(candidate) {
+    // Make sure all of the candidates are defined
+    if (!candidate) {
+      return;
+    }
+
+    // Each candidate gets its own div
+    var div = document.createElement('div');
+    div.setAttribute('role', 'presentation');
+    this.suggestionsContainer.appendChild(div);
+
+    var text, data, correction = false;
+    if (typeof candidate === 'string') {
+      if (candidate[0] === '*') { // it is an autocorrection candidate
+        candidate = candidate.substring(1);
+        correction = true;
+      }
+      data = text = candidate;
+    }
+    else {
+      text = candidate[0];
+      data = candidate[1];
+    }
+
+    var span = this.fitText(div, text, candidates.length);
+    span.setAttribute('role', 'option');
+    // TODO: the renderer should not be creating a business logic object,
+    // let's move it to somewhere else.
+    this.viewManager.registerView({
+      selection: true,
+      text: span.textContent,
+      data: data
+    }, {
+      element: span
```

Comment:
You should either `CandidatePanel#renderCandidatePanel` here or create a similar `LatinCandidatePanel#renderCandidatePanel` function here. `showCandidates` is way too long.